### PR TITLE
Add rotation to integration

### PR DESCRIPTION
**Pull request by fallenatlas**
# Description

Apply torque and angular impulses on integration to rotate the bodies.

Resources: https://research.ncl.ac.uk/game/mastersdegree/gametechnologies/physicstutorials/3angularmotion/Physics%20-%20Angular%20Motion.pdf

## Checklist

- [x] Self-review changes.
- [x] Evaluate impact on the documentation.
- [x] Add entry to the changelog's unreleased section.

### CODE REVIEW

**Review by roby2014 — APPROVED** [review]

**Comment by luishfonseca** [comment]
I should probably be removed from the list of reviewers during P1, I'll have more time again on P2 onwards

File: engine/include/cubos/engine/physics/components/inertia.hpp [hunk 1]
```diff
@@ -21,5 +22,19 @@ namespace cubos::engine
...
         bool autoUpdate = false;
+
+        glm::mat3 rotatedInertia(const glm::quat& rotationQuat) const
+        {
+            auto rotationMat = glm::mat3(rotationQuat);
+            return rotationMat * inertia * glm::transpose(rotationMat);
+        }
+
+        // TODO: check if we use this many times. If we do, it might be worth it to calculate and store them
+        // somewhere instead.
```
Comment (tomas7770) [inline engine/include/cubos/engine/physics/components/inertia.hpp:33]: Is this TODO supposed to exist?

**Review by tomas7770 — APPROVED** [review]
LGTM!

File: engine/include/cubos/engine/physics/components/inertia.hpp (hunk 1 above)
Comment (fallenatlas) [reply engine/include/cubos/engine/physics/components/inertia.hpp:33]: For now, yeah.

**Pull request merged**